hnn_core/tests/test_mpi_child.py

```diff
@@ -56,10 +131,19 @@ def test_empty_data():
 
 
 def test_data_len_mismatch():
-    """Test that an unexpected data length raises RuntimeError"""
-    data_bytes = b'\0'
-    expected_len = 2
+    """Test that padded data can be unpickled with warning for length """
+
+    with MPISimulation(skip_mpi_import=True) as mpi_sim:
+        pickled_bytes = mpi_sim._pickle_data({})
+
+    expected_len = len(pickled_bytes) + 1
+
     backend = MPIBackend()
-    with pytest.raises(RuntimeError, match="Failed to receive all data from "
-                       "the child MPI process. Expecting 2 bytes, got 1"):
-        backend._process_child_data(data_bytes, expected_len)
+    with pytest.warns(UserWarning) as record:
+        backend._process_child_data(pickled_bytes, expected_len)
+
+    expected_string = "Length of received data unexpected. " + \
+        "Expecting %d bytes, got %d" % (expected_len, len(pickled_bytes))
+
+    assert len(record) == 1
+    assert record[0].message.args[0] == expected_string
```

Comment:
use the `match` argument? I guess it's enough to check if the first part of the string matches?

Reply:
Sure, but why use a function if the operator does the same thing?

Reply:
I thought there was value in checking the entire string matched (I see your point if you only check the first part).

However, just like in the comment below, I don't think the test is necessary anymore!

Reply:
I'm leaving this in as the full string because we are checking that the mismatch between received and expected bytes. Unfortunately, with base64 this should stay in. If someone needs to debug it, it will be useful to know that there was a mismatch.